test/prog_tests/tpdu_test.py

```diff
@@ -0,0 +1,38 @@
+"""Unit test for tpdu class."""
+import pytest
+
+from xknx import XKNX
+from xknx.knxip.tpdu import TPDU
+from xknx.telegram import IndividualAddress, Telegram, TelegramDirection, TPDUType
+
+
+@pytest.mark.asyncio
+class TestTpdu:
+    """Test class for tpdu test."""
+
+    async def test_tpdu(self):
+        # test standard constructor
+        xknx = XKNX()
+        tpdu = TPDU(xknx, IndividualAddress("1.2.3"))
+        # add data from knx
+        in_bytes = bytes((0x11, 0, 0xB0, 0x60, 0, 0, 0x12, 0x01, 0, 0x80))
+        tpdu.from_knx(in_bytes)
+        # check serialzation result
+        o_bytes = tpdu.to_knx()
+        assert o_bytes == in_bytes
+        # test telegram property
+        telegram = Telegram(
+            IndividualAddress("1.2.3"), TelegramDirection.OUTGOING, None, None, TPDUType.T_CONNECT
+        )
+        tpdu.telegram = telegram
+        telegram_out = tpdu.telegram
+        assert telegram.destination_address == telegram_out.destination_address
+        # test constructor from telegram
+        tpdu = TPDU.init_from_telegram(xknx, telegram, IndividualAddress("1.2.3"))
+        # check serialzation result
+        o_bytes = tpdu.to_knx()
+        assert o_bytes == bytes((0x11, 0, 0xB0, 0x60, 0, 0, 0x12, 0x03, 0, 0x80))
```

Comment:
We prefer hex values over just bytes `bytes.fromhex()`.

To obtain the hex values just use `bytesobject.hex(" ")`.

Please apply it to all tests.